### PR TITLE
overlay: improve tests for TryNextPeer

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
Explicitely test for what `TryNextPeer` does

### CODE REVIEW

File: src/overlay/test/ItemFetcherTests.cpp [hunk 1]
```diff
@@ -221,41 +223,125 @@ TEST_CASE("ItemFetcher fetches", "[overlay][ItemFetcher]")
...
-                auto zeroEnvelope2 = makeEnvelope(0);
                 itemFetcher.fetch(zero, zeroEnvelope1);
-                itemFetcher.fetch(zero, zeroEnvelope2);
-            }
-            REQUIRE(asked.size() == 0);
 
-            while (asked.size() < 4)
-            {
-                clock.crank(true);
+                std::chrono::milliseconds const FETCH_NEXT_PEER{1500};
```
Comment (marta-lokhova) [inline src/overlay/test/ItemFetcherTests.cpp:270]: this breaks as soon as MS_TO_WAIT_FOR_FETCH_REPLY is changed. Why not just use that?

File: src/overlay/test/ItemFetcherTests.cpp [hunk 2]
```diff
@@ -221,41 +223,125 @@ TEST_CASE("ItemFetcher fetches", "[overlay][ItemFetcher]")
...
+
+                auto testNotFound = [&](bool respond) {
+                    int constexpr ITERATIONS = 100;
+                    for (auto i = ITERATIONS; i != 0; --i)
```
Comment (marta-lokhova) [inline src/overlay/test/ItemFetcherTests.cpp:274]: is this loop to test what happens when we run out of peers? curious why we need so many iterations here

**Review by marta-lokhova — COMMENTED** [review]
added a few suggestions

File: src/overlay/test/ItemFetcherTests.cpp [hunk 3]
```diff
@@ -221,41 +223,125 @@ TEST_CASE("ItemFetcher fetches", "[overlay][ItemFetcher]")
...
+                    VirtualClock::time_point prevGroup;
+
+                    for (size_t i = 0; i < asked.size(); i += 2)
+                    {
+                        auto refTP = askedTP[i];
+                        auto delta = askedTP[i + 1] - refTP;
+                        // record time when alternating between peers
+                        if (respond)
+                        {
+                            // this could probably lowered a lot more as we're
```
[the remark this answers is not in the document]
Comment (MonsieurNicolas) [reply src/overlay/test/ItemFetcherTests.cpp]: I ended up reworking those tests to be easier to understand. Let me know what you think!

**Comment by MonsieurNicolas** [comment]
@marta-lokhova can you merge this? I think I addressed your comments some time ago

**Comment by marta-lokhova** [comment]
r+ 79e445e07d3698dcf2ff3c92212341b1b64b9363

**Pull request merged**